src/lib.rs

```diff
@@ -448,6 +459,45 @@ macro_rules! whatever {
     };
 }
 
+/// Ensure a condition is true. If it is not, return a stringly-typed
+/// error message.
+///
+/// This can be used with the provided [`Whatever`][] type or with a
+/// custom error type that uses `snafu(whatever)`.
+///
+/// ## Examples
+///
+/// ```rust
+/// use snafu::prelude::*;
+///
+/// #[derive(Debug, Snafu)]
+/// #[snafu(whatever, display("Error was: {message}"))]
+/// struct Error {
+///     message: String,
+/// }
+/// type Result<T, E = Error> = std::result::Result<T, E>;
+///
+/// fn get_bank_account_balance(account_id: &str) -> Result<u8> {
+/// # fn moon_is_rising() -> bool { false }
+///     ensure_whatever!(
+///         moon_is_rising(),
```

Comment:
Was it intended to negate the condition?  If I'm not mistaken, this changes it from `if moon_is_rising()` to (after macro expansion) `if !moon_is_rising()`.

Reply:
Not intended, *per se*, but I’m ok with it. Since it’s just an example, whatever shows the feature easiest is a good choice, IMO. I don’t expect anyone will remember the example from one version to the next 😅